undertow/src/main/java/org/wildfly/extension/undertow/deployment/JBossWebParsingDeploymentProcessor.java

```diff
@@ -74,6 +74,11 @@ public void deploy(DeploymentPhaseContext phaseContext) throws DeploymentUnitPro
                 warMetaData.setJBossWebMetaData(jBossWebMetaData);
                 // if the jboss-web.xml has a distinct-name configured, then attach the value to this
                 // deployment unit
+                if(jBossWebMetaData.getValves() != null) {
+                    jBossWebMetaData.getValves().stream().forEach((valve) -> {
```

Comment:
At the very least, note that `Iterable` has `forEach` as well, so you don't really have to create a `Stream`. Though I'd personally agree with Tomaz here.